lib/Targeting/targeting_manager.dart

```diff
@@ -0,0 +1,206 @@
+import 'package:flagship/model/trageting.dart';
+
+/// :nodoc:
+const String FS_USERS = "fs_users";
+
+enum FSOperator {
+  EQUALS,
+
+  NOT_EQUALS,
+
+  GREATER_THAN,
+
+  GREATER_THAN_OR_EQUALS,
+
+  LOWER_THAN,
+
+  LOWER_THAN_OR_EQUALS,
+
+  CONTAINS,
+
+  NOT_CONTAINS,
+
+  Unknown
+}
+
+class TargetingManager {
+  String userId = "";
+  Map<String, dynamic> currentContext = {};
+
+  TargetingManager(this.userId, this.currentContext) {
+    // add fs_all_users
+    this.currentContext["fs_all_users"] = "";
+  }
+
+  bool isTargetingGroupIsOkay(Targeting? targeting) {
+    // Groupe de variations,
+    List<bool> booleanResultGroup = [];
+    for (TargetingGroup itemGroup in targeting?.targetingGroups ?? []) {
+      booleanResultGroup.add(checkTargetGroupIsOkay(itemGroup));
+    }
+
+    return booleanResultGroup.contains(true);
+  }
+
+  bool checkTargetGroupIsOkay(TargetingGroup itemTargetGroup) {
+    for (ItemTarget itemTarget in itemTargetGroup.targetings) {
+      dynamic currentContextValue = getCurrentValueFromCtx(itemTarget.tragetKey);
+      // Audience value
+      dynamic audienceValue = itemTarget.targetValue;
+
+      // Create the type operator
+      FSOperator opType = createOperator(itemTarget.targetOperator);
+
+      // Special treatment for array
+      bool isOkay = false;
+
+      if ((audienceValue is List<String>) ||
+          (audienceValue is List<int>) ||
+          (audienceValue is List<double>) ||
+          (audienceValue is List<dynamic>)) {
+        isOkay = checkTargetingForList(currentContextValue, opType, audienceValue);
+      } else {
+        isOkay = checkCondition(currentContextValue, opType, audienceValue);
+      }
+
+      if (isOkay == false) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  bool checkTargetingForList(dynamic currentValue, FSOperator opType, List<dynamic> listAudience) {
+    // Chekc the type list before
+    bool isOkay = false;
+    int result = 0;
```

Comment:
Feels weird to use an integer here, since we only use is to check if it is still 0. I guess we could use a boolean, no?

Reply:
Also, it would be nice to have a better wording for that variable

Reply:
Okay.